mater/lib/src/stores/filestore.rs

```diff
@@ -183,11 +270,117 @@ mod test {
+
+        // Create test file with random data to ensure unique chunks
+        let mut rng = thread_rng();
+        let test_data = (0..512 * 1024).map(|_| rng.gen::<u8>()).collect::<Vec<_>>();
+
+        trace!("Creating test file of size: {} bytes", test_data.len());
+        tokio::fs::write(&input_path, &test_data).await.unwrap();
+
+        let source_file = File::open(&input_path).await.unwrap();
+        let output_file = File::create(&temp_path).await.unwrap();
+
+        let config = Config::balanced_unixfs(64 * 1024, 2);
+
+        let root_cid = create_filestore(source_file, output_file, config)
+            .await
+            .unwrap();
+        trace!("Root CID: {}", root_cid);
+
+        // Read back and verify structure
+        let file = File::open(&temp_path).await.unwrap();
+        let mut reader = crate::CarV2Reader::new(file);
+
+        reader.read_pragma().await.unwrap();
+        reader.read_header().await.unwrap();
+        reader.read_v1_header().await.unwrap();
+
+        // Track all unique blocks and statistics
+        let mut unique_blocks = HashSet::new();
+        let mut leaf_blocks = HashSet::new();
+        let mut parent_blocks = HashSet::new();
+        let mut level_sizes = Vec::new();
+        let mut current_level_nodes = HashSet::new();
+        let mut current_level = 0;
+
+        while let Ok((cid, data)) = reader.read_block().await {
+            unique_blocks.insert(cid);
+
+            let pb_node: PbNode = DagPbCodec::decode(&data[..]).unwrap();
+            let reader =
+                &mut quick_protobuf::BytesReader::from_bytes(&pb_node.data.clone().unwrap());
+            let bytes = &pb_node.data.unwrap();
+            let unixfs_data = Data::from_reader(reader, bytes).unwrap();
+
+            if pb_node.links.is_empty() {
+                leaf_blocks.insert(cid);
+                trace!("Found leaf node: {} (size: {})", cid, data.len());
+                trace!(
+                    "  Data size: {}",
+                    unixfs_data.Data.as_ref().map_or(0, |d| d.len())
+                );
+                trace!("  Blocksizes: {:?}", unixfs_data.blocksizes);
+
+                // New level if this is first leaf
+                if current_level_nodes.is_empty() {
+                    level_sizes.push(0);
+                    current_level = level_sizes.len() - 1;
+                }
+            } else {
+                parent_blocks.insert(cid);
+
+                trace!(
+                    "Found parent node: {} with {} links (size: {})",
+                    cid,
+                    pb_node.links.len(),
+                    data.len()
+                );
+                trace!("  Total filesize: {:?}", unixfs_data.filesize);
+                trace!("  Blocksizes: {:?}", unixfs_data.blocksizes);
+
+                for link in &pb_node.links {
+                    trace!("  -> Link to: {} (size: {:?})", link.cid, link.size);
+                }
```

Comment:
This loop just performs a log and can't be optimized out without compilation flags. Remove
```suggestion
```